Lib/test/test_importlib/test_api.py

```diff
@@ -361,6 +361,20 @@ def test_reload_submodule(self):
             reloaded = self.init.reload(ham)
             self.assertIs(reloaded, ham)
 
+    def test_reload_faliure(self):
+        #Test that reload throws ModuleNotFounderror if reloading
+        #a module whose spec cannot be found.
+        name = 'spam'
+        subname = 'ham'
+        with test_util.temp_module(name, pkg=True) as pkg_dir:
+            fullname, _ = test_util.submodule(name, subname, pkg_dir)
+            module = self.init.import_module(fullname)
+            module.__spec__ = None
+            module.__name__ = 'destroyed_spam'
+            sys.modules[module.__name__] = module
```

Comment:
Not changing the name won't work. If I don't modify the name, the method _find_spec inside importlib/_bootstrap.py tries to find spec from three finder from sys.meta_path:  _frozen_importlib.BuiltinImporter, _frozen_importlib.FrozenImporter and _frozen_importlib_external.PathFinder respectively. But since I have not changed the name and the module by that filename still exists PathFinder can find the spec (reload the spec) from the file. Hence I do need to change the name and put it explicitly in sys.modules